Acrolinx.Sidebar/AcrolinxSidebar.cs

```diff
@@ -1,698 +1,715 @@
+        public string StartPageSourceLocation
+        {
+            get
+            {
+                return startPageLocation;
+            }
+
+            set
+            {
+                startPageLocation = value;
+            }
+        }
+
+        public enum SoftwareComponentCategory
+        {
+
+            /// <summary>
+            /// There should be exactly one MAIN component.
+            /// This information is used to identify your client on the server.
+            /// Version information about this components might be displayed more prominently.
+            /// </summary>
+            MAIN,
+
+            /// <summary>
+            /// Version information about such components are displayed in the about dialog.
+            /// </summary>
+
+            DEFAULT,
+            /// <summary>
+            /// Version information about such components are displayed in the detail section of the about dialog or not at all.
+            /// </summary>
+            DETAIL
+        }
+        public void RegisterClientComponent(Assembly assembly, string humanReadableName, SoftwareComponentCategory category)
+        {
+            Contract.Requires(assembly != null);
+            Contract.Requires(humanReadableName != null);
+            if (assembly != null)
+            {
+                var name = assembly.GetName();
+                RegisterClientComponent(name.Name, humanReadableName, name.Version.ToString(), category);
+            }
+        }
+
+        private bool mainComponentSet = false;
+        public void RegisterClientComponent(string id, string name, string version, SoftwareComponentCategory category)
+        {
+            Contract.Requires(id != null);
+
+            mainComponentSet |= category == SoftwareComponentCategory.MAIN;
+
+            var clientComponents = InitParameters["clientComponents"].Value<JArray>();
+
+            var component = new JObject();
+            component.Add("id", id.ToLower());
+            component.Add("name", name);
+            component.Add("version", version);
+            component.Add("category", category.ToString().ToUpper());
+            if (category != SoftwareComponentCategory.DETAIL)
+            {
+                clientComponents.Add(component);
+            }
+            Logger.AcroLog.Info(name + "\t" + version);
+        }
+
+        public String Check(IDocument document)
+        {
+            acrolinxPlugin.Document = document;
+            Logger.AcroLog.Debug("Content: " + document.Content);
+
+            var code = "new function(){var c = window.external.getContent(); "
+                + "return acrolinxSidebar.checkGlobal(c, {inputFormat:'" + document.Format.ToString().ToUpper() + "', requestDescription:{documentReference: '"
+                + document.Reference.Replace("\\", "\\\\").Replace("'", "\\'").Replace("\n", "").Replace("\r", "") + "'}, selection:{ranges:" + SerializeSelection(document.Selections) + "}})}();";

```

Comment:
@mscc-sascha Can you also use the the function GetJavaScriptFriendlyParameterString here? As its doing the same here.